src/search/pdbs/match_tree.h

```diff
@@ -22,11 +24,9 @@ namespace pdbs {
 */
 
 class MatchTree {
-    TaskProxy task_proxy;
+    const TaskProxy &task_proxy;
```

Comment:
Do we need a reference here? `TaskProxy` should be light-weight enough to copy and that would reduce the complexity. For example, in `Heuristic` we also store a `TaskProxy` not a reference. I think we agreed at some point to use const references for passing around proxies but I don't think they are necessary for storage.